src/libraries/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -143,7 +140,7 @@ public FileAttributes Attributes
         public DateTimeOffset LastAccessTimeUtc => _status.GetLastAccessTime(FullPath, continueOnError: true);
         public DateTimeOffset LastWriteTimeUtc => _status.GetLastWriteTime(FullPath, continueOnError: true);
         public bool IsDirectory => _status.InitiallyDirectory;
-        public bool IsHidden => _directoryEntry.Name[0] == '.';
+        public bool IsHidden => _directoryEntry.Name[0] == '.' || (Attributes & FileAttributes.Hidden) != 0;
```

Comment:
Actually, I think this should be:
```cs
        public bool IsHidden => _directoryEntry.Name[0] == '.' || (Interop.Sys.CanSetHiddenFlag && (Attributes & FileAttributes.Hidden) != 0);
```
Otherwise, if the user sets the Hidden flag in Linux, it won't really hide the file, but this will return true.
I'm going to test this on Linux.

Reply:
I added this check too, just in case. I noticed that the Hidden flag cannot be set to the file Attributes in Linux, because the property performs some checks to see if it's possible. But just in case, attempting to read the value should be protected.

Reply:
The public property was introduced here: https://github.com/dotnet/runtime/commit/c60340730ce3cb5002001e505e1fb8b360496771

But the change that added the case for UF_HIDDEN should've added this check:
https://github.com/dotnet/corefx/pull/34560

Reply:
I had a conversation with @Jozkee about this and the `Interop.Sys.CanSetHiddenFlag` check is not necessary, because the Linux P/Invoke will never allow setting it. So I'm reverting the last commit.

Reply:
`Attributes` adds `FileAttributes.ReadOnly` which is irrelevant in this operation.
```suggestion
        public bool IsHidden => _directoryEntry.Name[0] == '.' || (_initialAttributes & FileAttributes.Hidden) != 0;
```

Reply:
Can you please also add a comment on top explaining where the attribute is supported?

Reply:
Suggestion applied, and comment added.